lib/traject/macros/custom.rb

```diff
@@ -205,12 +205,22 @@ def self.includes_oclc_indicators?(sf_a)
           sf_a.include?('OCLC')
       end
 
-      # Extract ht_id
-      def extract_ht_id
+      def extract_hathi_data
         lambda do |_record, accumulator, context|
           oclc_number = context.output_hash&.dig('oclc_number_ssim')&.first
-          accumulator << HATHI_ETAS_OVERLAP[oclc_number]
-          accumulator.compact
+          ht_hash = HATHI_MULTI_OVERLAP&.dig(oclc_number)&.first || HATHI_MONO_OVERLAP&.dig(oclc_number)&.first
+          accumulator << ht_hash.to_json unless ht_hash.nil?
+        end
+      end
+
+      def hathi_to_hash(ht_format)
```

Comment:
Can you add a spec to cover this method?

I am really having a hard time with line 219 especially. It's using a ["pretzel colon"](https://www.honeybadger.io/blog/how-ruby-ampersand-colon-works/) to do the following:

"create a new array that is grouped by each value in the array with the first element removed"

Is that understanding correct?

Reply:
That line will group the records by oclc by removing the oclc from the array:

```
# [[1, 10, allow] [1, 20, deny], [2, 30, allow], [2, 40, deny]]
result = result.group_by(&:shift)
# {1 => [[10, allow], [20, deny]], 2 => [[30, allow], [40, deny]]}
#  ^ oclc                          ^ oclc
```

Reply:
I intended to have this method as private so I did not add a spec for it (I think the other specs kinda covering it) 🤔 
I just wasn't sure where the better place to put this method would be (here?), wanted to see what your thoughts are on it.

Reply:
specs added for this method

Reply:
ok I had it wrong, it should have been "create a new _hash_  of arrays that is grouped by (i.e. keyed by) the first value of each array (the oclc number) not including the oclc number in the array" Pretty cool!

Reply:
right ☝️